Helper/Data.php

```diff
@@ -1184,9 +1192,18 @@ public function buildRequestHeaders($payment = null)
             HeaderDataBuilderInterface::MERCHANT_APPLICATION_VERSION  => $this->getModuleVersion()
         ];
 
-        if(isset($payment) && !is_null($payment->getAdditionalInformation(HeaderDataBuilderInterface::ADDITIONAL_DATA_FRONTEND_TYPE_KEY))) {
-            $headers[HeaderDataBuilderInterface::EXTERNAL_PLATFORM_FRONTEND_TYPE] =
-                $payment->getAdditionalInformation(HeaderDataBuilderInterface::ADDITIONAL_DATA_FRONTEND_TYPE_KEY);
+        if (isset($payment)) {
+            $frontendType = $payment->getAdditionalInformation(HeaderDataBuilderInterface::ADDITIONAL_DATA_FRONTEND_TYPE_KEY);
+            if (is_null($frontendType)) {
+                // Check the request URI
+                $requestUri = $this->request->getPathInfo();
+                if (str_contains($requestUri, '/graphql')) {
+                    $frontendType = 'headless-graphql';
+                } else {
+                    $frontendType = 'headless-rest';
```

Comment:
Express payments will fall under `headless-rest` category unless we indicate the `frontendType` in `additional_data` of place order functions of express method renderers. This might require some data clean-up to remove express payments from `headless-rest` payments. However, it would become cumbersome after having a huge data set. We can merge this as it is but until fixing it on express payment module, we will get some dirty data. Could you please check this with product?